source/dhtslib/memory.d

```diff
@@ -0,0 +1,92 @@
+module dhtslib.memory;
+
+import std.traits : isPointer, isSomeFunction, ReturnType;
+import core.lifetime : move;
+import std.typecons : RefCounted, RefCountedAutoInitialize;
+import htslib.sam : bam1_t, bam_hdr_t, bam_destroy1, bam_hdr_destroy;
+import htslib.vcf : bcf1_t,bcf_hdr_t, bcf_destroy, bcf_hdr_destroy;
+import htslib.hts : htsFile, hts_close;
+import htslib.hts_log;
+
+/// Template struct that performs reference
+/// counting on htslib pointers and destroys with specified function
+struct HtslibMemory(T, alias destroy)
+if(!isPointer!T && isSomeFunction!destroy)
+{
+    @safe:
+    /// Pointer Wrapper
+    struct HtslibPtr
```

Comment:
you probably want to make this a `static struct` because it doesn't depend on anything from `HtslibMemory` (template parameters don't count)

Reply:
done d26b393.